### PR TITLE
f-contact-preferences@v0.5.2 - Added a mapper module

### PR DESCRIPTION
**Pull request by oliversweb**
### Added
- Added contact preferences mapper & unit tests.

### CODE REVIEW

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import {
+    NON_EDITABLE_PREFERENCE_KEYS,
+    NON_VISIBLE_PREFERENCE_KEYS
+} from '../../constants';
+
+export const mapToPreferencesViewModel = ({ Preferences, PreferenceVersionViewed }) => {
```
Comment (zlatinivanov-je) [inline packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js:12]: Just a little unclear why the arguments are wrapped in curlies `{ Preferences, PreferenceVersionViewed }`

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const model = {
+        preferences: Preferences.map(({
+            DisplayName, Sort, Key, Push, Email, Sms
```
Comment (zlatinivanov-je) [inline packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js]: Also is there a requirement for the keys to start with a capital letter?

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js (hunk 1 above)
Comment (zlatinivanov-je) [reply packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js:12]: Possibly JSDoc comments can help

File: packages/components/pages/f-contact-preferences/test-utils/setup.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    DeviceType: null,
+    PhoneNumber: null,
+    PreferenceVersionViewed: 0
+};
+
+export const baseUrl = 'https://smartGatewayBaseUrl.com';
+
+export const token = 'some-auth-token';
+
+export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';
```
Comment (zlatinivanov-je) [inline packages/components/pages/f-contact-preferences/test-utils/setup.js:125]: Do we have any concerns around that value being public?

**Review by zlatinivanov-je — COMMENTED** [review]
Generally looks good to me, just a couple of questions.

File: packages/components/pages/f-contact-preferences/test-utils/setup.js (hunk 3 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/test-utils/setup.js:125]: No, just a random guid for tests.

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js (hunk 2 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js]: These are the actual fields inside the api response, so not internal fields

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js (hunk 1 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js:12]: Object destructuring (i.e. a single object is passed and I extract & assign in a single line). I will add a comment 👍

Comment (zlatinivanov-je) [reply packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js:12]: Thank you for explaining and adding the JSDocs. 
Would you take a suggestion that may make the intent a little clearer in my view. Feel free to object if you feel strongly about that.
 
```suggestion
export const mapToPreferencesViewModel = (preferencesResponseData) => {
    const { Preferences, PreferenceVersionViewed } = preferencesResponseData; 
```

Comment (DamianMullins) [reply packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js:12]: Just throwing in my 2-cents! Personally, I find the code is clearer with the inline destructure as it is currently (I do it in menu in quite a few places), I guess it's down to personal preference?

Comment (zlatinivanov-je) [reply packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js:12]: Happy to approve if that is the consensus here. Thank you @DamianMullins

**Review by zlatinivanov-je — DISMISSED** [review]

File: packages/components/pages/f-contact-preferences/CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 
+v0.5.2
+------------------------------
+*November 15, 2021*
+
+### Added
+- Added mapper.
```
Comment (kevinrodrigues) [inline packages/components/pages/f-contact-preferences/CHANGELOG.md]: May be worth adding a tiny bit more detail here? `added contact preferences mapper & unit tests`? :)

File: packages/components/pages/f-contact-preferences/src/services/mapping/_tests/contactPreferences.mapper.test.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+            // Act
+            const actual = mapToPreferencesUpdateModel(contactPreferencesViewModel);
+
+            // Assert
+            expect(actual).toEqual(expected);
+        });
+    });
+
+    describe('When calling `filterSortPreferences`', () => {
```
Comment (kevinrodrigues) [inline packages/components/pages/f-contact-preferences/src/services/mapping/_tests/contactPreferences.mapper.test.js:41]: Does this need to cover the return of the empty array when say `preferences` do not exist?

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js (hunk 2 above)
Comment (kevinrodrigues) [reply packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js]: Would you be able to line break each key so it reads like:

```
DisplayName,
Sort,
Key....

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Maps the view model to the Body payload for the api update request.
+ * @param {object} preferences - The preferences contained in the passing object parameter (i.e. state.preferences).
+ * @param {object} preferenceVersionViewed - The version value contained in the passing object parameter (i.e. state.preferenceVersionViewed).
+ */
+export const mapToPreferencesUpdateModel = ({ preferences, preferenceVersionViewed }) => {
+    const model = {
+        Preferences: preferences.map(({
+            displayName, sort, key, pushValue, emailValue, smsValue
```
Comment (kevinrodrigues) [inline packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js]: Similar, feel it reads nicer but happy to leave these as is if people prefer.

File: packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        DeviceToken: null,
+        DeviceType: null,
+        PhoneNumber: null,
+        PreferenceVersionViewed: preferenceVersionViewed
+    };
+
+    return model;
+};
+
+export const filterSortPreferences = preferences => {
```
Comment (kevinrodrigues) [inline packages/components/pages/f-contact-preferences/src/services/mapping/contactPreferences.mapper.js:78]: Could you add a jsDoc to this method please.

**Review by kevinrodrigues — APPROVED** [review]

**Review by zlatinivanov-je — APPROVED** [review]

**Pull request merged**